src/python/zquantum/qaoa/maxcut.py

```diff
@@ -11,21 +11,25 @@
 
 
 def create_farhi_qaoa_circuits(
-    hamiltonians: List[QubitOperator], number_of_layers: int
+    hamiltonians: List[QubitOperator], number_of_layers: Union[int, List[int]]
 ):
-    """Creates parameterizable quantum circuits based on the farhi qaoa ansatz for each 
-    hamiltonian in the input list using the set number of layers. 
+    """Creates parameterizable quantum circuits based on the farhi qaoa ansatz for each
+    hamiltonian in the input list using the set number of layers.
 
     Args:
         hamiltonians (List[QubitOperator]): List of hamiltonians for constructing the
             circuits
         number_of_layers (int): The number of layers of the ansatz in the circuit
-                
+
     Returns:
         List of zquantum.core.circuit.Circuit
     """
+    if isinstance(number_of_layers, int):
+        number_of_layers = [number_of_layers for _ in range(len(hamiltonians))]
+    number_of_layers_list = number_of_layers
+
     circuitset = []
-    for hamiltonian in hamiltonians:
+    for number_of_layers, hamiltonian in zip(number_of_layers_list, hamiltonians):
```

Comment:
If the `number_of_layers_list` is of a different length than `hamiltonians`, this will fail.
I think we should assert that earlier so that the error message is clearer to the user.

Reply:
Addressed here: d7a5362